server/download-file.js

```diff
@@ -0,0 +1,23 @@
+const { pipeline } = require('stream');
+const { promisify } = require('util');
+const streamPipeline = promisify(pipeline);
+
+async function downloadFileHandler(req, res) {
+  const { url } = req.query;
+  if (!url) {
+    return res.status(400).json({ error: 'Missing url parameter' });
+  } else if (!/https:\/\/opencollective-(production|staging)\.s3[.-]us-west-1\.amazonaws\.com/.test(url)) {
+    return res.status(400).json({ error: 'Only files from Open Collecive are allowed' });
+  }
+  const response = await fetch(url);
```

Comment:
@Betree what do you think? Since we're making sure the url follows the pattern of our s3 buckets or the transactions csv rest API endpoint, I'm thinking we're fine here?

Reply:
We need to straighten that regex, it can currently be bypassed with something like:

> https://example.com/malicious-code?url=https://opencollective-staging.s3.us-west-1.amazonaws.com/expense-item/74dfb507-2691-4d6a-a233-37b0f170e130/small.jpg

I can recommend something like:

```es6
const parsedUrl = new URL(req.query.url);
if (!/^https:\/\/opencollective-(production|staging)\.s3[.-]us-west-1\.amazonaws\.com/.test(parsedUrl.origin)) {
  return res.status(400).json({ error: 'Only files from Open Collecive are allowed' });
}
```

It would also be great to make it env dependent - don't let people use this on staging with prod URLs & vis & versa.

Ultimately, if possible, it would be great to have this in the API, as we'll also be able to handle access control from there (https://github.com/opencollective/opencollective/issues/6351).